src/nvim/api/keysets.lua

```diff
@@ -0,0 +1,19 @@
+return {
+  set_extmark = {
```

Comment:
should it be `buf_set_extmark`? Or even the full api name `nvim_buf_set_extmark` , to avoid cognitive indirection.

Reply:
Not sure, but they are not public and not necessarily tied to one specific function, for instance we have two entry points for float_config.

Reply:
ah right, these are more like types, not signatures

Reply:
Currently all key sets are for Object structs, but they could also latter be used for other stuff, like method lookup tables (and replace gperf, hint hint).

Reply:
> (and replace gperf, hint hint

read my mind